### PR TITLE
envoy: replace connect_authority with set_filter_state

### PR DESCRIPTION
**Pull request by kyessenov**
Change-Id: Ie027edd6f589b6a4d869ffe9b1a69ed8490a585e

Follow up to https://github.com/istio/proxy/pull/5032

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by kyessenov** [comment]
Yay! One more filter gone.

**Review by GregHanson — COMMENTED** [review]
Do we need release notes for this and the related one that removed the istio set_internal_dst_address?  EnvoyFilters that rely on name matching for either MERGE or INSERT is the one scenario off the top of my head.  Any other scenario that might surface during an upgrade?

**Comment by kyessenov** [comment]
Those are all experimental ambient filters. They are not used for sidecars AFAICT.

File: pilot/pkg/xds/filters/filters.go [hunk 1]
```diff
@@ -301,27 +304,70 @@ var (
 	ConnectAuthorityFilter = &hcm.HttpFilter{
 		Name: "connect_authority",
 		ConfigType: &hcm.HttpFilter_TypedConfig{
-			TypedConfig: protoconv.TypedStruct("type.googleapis.com/io.istio.http.connect_authority.Config"),
+			TypedConfig: protoconv.MessageToAny(&sfs.Config{
+				OnRequestHeaders: []*sfsvalue.FilterStateValue{
+					{
+						Key: &sfsvalue.FilterStateValue_ObjectKey{
+							ObjectKey: "envoy.filters.listener.original_dst.local_ip",
```
Comment (GregHanson) [inline pilot/pkg/xds/filters/filters.go:311]: @kyessenov is there a reason this ObjectKey has to be tied to a `well_known_filter_state`?  I noticed I get this [error](https://github.com/envoyproxy/envoy/blame/5228bf8f080358a00f3996d3c3a68f12f5419b1e/source/extensions/filters/common/set_filter_state/filter_config.cc#L22) if I try to set an arbitrary key value.

I don't think this was a requirement in the original `decoder_callbacks_->streamInfo().filterState()->setData(<key>,...)` function in proxy - is there a reason for this regression?

Comment (kyessenov) [reply pilot/pkg/xds/filters/filters.go:311]: It was left out from the initial version. The reason is you need to look up the C++ object type via a key.

Comment (GregHanson) [reply pilot/pkg/xds/filters/filters.go:311]: >initial version

So there are plans to support this in the near term?

Comment (kyessenov) [reply pilot/pkg/xds/filters/filters.go:311]: Yes, string and integer keys were requested.

Comment (GregHanson) [reply pilot/pkg/xds/filters/filters.go:311]: ok, I see the discussion:
https://github.com/envoyproxy/envoy/pull/29844#discussion_r1355812830

thanks for clarifying

**Pull request merged**